api/client/build.go

```diff
@@ -265,6 +268,18 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	headers.Add("X-Registry-Config", base64.URLEncoding.EncodeToString(buf))
 	headers.Set("Content-Type", "application/tar")
 
+	// collect all the build-time environment variables for the container
+	buildArgs := runconfig.ConvertKVStringsToMap(flArg.GetAll())
+	buf, err = json.Marshal(buildArgs)
+	if err != nil {
+		return err
+	}
+	headers.Add("X-Docker-BuildArgs", base64.URLEncoding.EncodeToString(buf))
+
+	if context != nil {
```

Comment:
Somehow missed this comment. yes, it's not part of this patch. It might have come from code change in this area and rebasing my diffs over them. I will remove it.